### ISSUE TITLE
Support for setting custom maximum requests per second

### ISSUE THREAD

**Issue by soredake** [dkhamsing/awesome_bot#123]
something like `--max-requests` to prevent getting false 403 on some sites.

**Comment by dkhamsing** [comment]
Can you provide an example for a site and a number of request ?

**Comment by soredake** [comment]
https://chan.sankakucomplex.com
Typical log:
``` 
  781. [L1500] 429 https://chan.sankakucomplex.com/post/show/82338
  782. [L1501] 429 https://chan.sankakucomplex.com/post/show/828541
  783. [L1502] 429 https://chan.sankakucomplex.com/post/show/847569
  784. [L1503] 429 https://chan.sankakucomplex.com/post/show/859296
  785. [L1504] 429 https://chan.sankakucomplex.com/post/show/945883
  786. [L1505] 429 https://chan.sankakucomplex.com/post/show/947875
  787. [L1506] 429 https://chan.sankakucomplex.com/post/show/949241
  788. [L1507] 429 https://chan.sankakucomplex.com/post/show/965668
  789. [L1508] 429 https://chan.sankakucomplex.com/post/show/992332
  790. [L1509] 429 https://chan.sankakucomplex.com/post/show/998051
```

**Comment by dkhamsing** [comment]
I'm working on a different solution that may be a little simpler
Can you send me the readme or file with links for me to test

**Comment by dkhamsing** [comment]
Uhm nevermind

**Comment by dkhamsing** [comment]
Without fix, run time 0m1.556s

```
./awesome_bot max-req 
> Checking links in max-req
Links to check: 50
  01. https://chan.sankakucomplex.com/post/show/82338 
  02. https://chan.sankakucomplex.com/post/show/828541 
  03. https://chan.sankakucomplex.com/post/show/847569 
  04. https://chan.sankakucomplex.com/post/show/859296 
  05. https://chan.sankakucomplex.com/post/show/945883 
  06. https://chan.sankakucomplex.com/post/show/947875 
  07. https://chan.sankakucomplex.com/post/show/949241 
  08. https://chan.sankakucomplex.com/post/show/965668 
  09. https://chan.sankakucomplex.com/post/show/992332 
  10. https://chan.sankakucomplex.com/post/show/998051 
  11. https://chan.sankakucomplex.com/post/show/82323 
  12. https://chan.sankakucomplex.com/post/show/82841 
  13. https://chan.sankakucomplex.com/post/show/84769 
  14. https://chan.sankakucomplex.com/post/show/829296 
  15. https://chan.sankakucomplex.com/post/show/925883 
  16. https://chan.sankakucomplex.com/post/show/927875 
  17. https://chan.sankakucomplex.com/post/show/929241 
  18. https://chan.sankakucomplex.com/post/show/925668 
  19. https://chan.sankakucomplex.com/post/show/922332 
  20. https://chan.sankakucomplex.com/post/show/928051 
  21. https://chan.sankakucomplex.com/post/show/8233 
  22. https://chan.sankakucomplex.com/post/show/8285 
  23. https://chan.sankakucomplex.com/post/show/8475 
  24. https://chan.sankakucomplex.com/post/show/8592 
  25. https://chan.sankakucomplex.com/post/show/9458 
  26. https://chan.sankakucomplex.com/post/show/9478 
  27. https://chan.sankakucomplex.com/post/show/9492 
  28. https://chan.sankakucomplex.com/post/show/9656 
  29. https://chan.sankakucomplex.com/post/show/9923 
  30. https://chan.sankakucomplex.com/post/show/9980 
  31. https://chan.sankakucomplex.com/post/show/8281 
  32. https://chan.sankakucomplex.com/post/show/8542 
  33. https://chan.sankakucomplex.com/post/show/847562 
  34. https://chan.sankakucomplex.com/post/show/85992 
  35. https://chan.sankakucomplex.com/post/show/9482 
  36. https://chan.sankakucomplex.com/post/show/9472 
  37. https://chan.sankakucomplex.com/post/show/949242 
  38. https://chan.sankakucomplex.com/post/show/96562 
  39. https://chan.sankakucomplex.com/post/show/992333 
  40. https://chan.sankakucomplex.com/post/show/9053 
  41. https://chan.sankakucomplex.com/post/show/82138 
  42. https://chan.sankakucomplex.com/post/show/821541 
  43. https://chan.sankakucomplex.com/post/show/841569 
  44. https://chan.sankakucomplex.com/post/show/851296 
  45. https://chan.sankakucomplex.com/post/show/941883 
  46. https://chan.sankakucomplex.com/post/show/941875 
  47. https://chan.sankakucomplex.com/post/show/941241 
  48. https://chan.sankakucomplex.com/post/show/961668 
  49. https://chan.sankakucomplex.com/post/show/991332 
  50. https://chan.sankakucomplex.com/post/show/991051 
Checking URLs: ✓✓✓✓✓✓xxx✓x✓✓✓xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

Issues :-(
> Links 
  01. [L01] 429 https://chan.sankakucomplex.com/post/show/8233  
  02. [L02] 429 https://chan.sankakucomplex.com/post/show/8285  
  03. [L03] 429 https://chan.sankakucomplex.com/post/show/8475  
  04. [L04] 429 https://chan.sankakucomplex.com/post/show/8592  
  05. [L05] 429 https://chan.sankakucomplex.com/post/show/9458  
  06. [L06] 429 https://chan.sankakucomplex.com/post/show/9478  
  07. [L07] 429 https://chan.sankakucomplex.com/post/show/9492  
  08. [L08] 429 https://chan.sankakucomplex.com/post/show/9656  
  09. [L09] 429 https://chan.sankakucomplex.com/post/show/9923  
  10. [L10] 429 https://chan.sankakucomplex.com/post/show/9980  
  11. [L12] 429 https://chan.sankakucomplex.com/post/show/82323  
  12. [L13] 429 https://chan.sankakucomplex.com/post/show/82841  
  13. [L14] 429 https://chan.sankakucomplex.com/post/show/84769  
  14. [L15] 429 https://chan.sankakucomplex.com/post/show/829296  
  15. [L16] 429 https://chan.sankakucomplex.com/post/show/925883  
  16. [L17] 429 https://chan.sankakucomplex.com/post/show/927875  
  17. [L18] 429 https://chan.sankakucomplex.com/post/show/929241  
  18. [L19] 429 https://chan.sankakucomplex.com/post/show/925668  
  19. [L20] 429 https://chan.sankakucomplex.com/post/show/922332  
  20. [L21] 429 https://chan.sankakucomplex.com/post/show/928051  
  21. [L34] 429 https://chan.sankakucomplex.com/post/show/8281  
  22. [L35] 429 https://chan.sankakucomplex.com/post/show/8542  
  23. [L36] 429 https://chan.sankakucomplex.com/post/show/847562  
  24. [L37] 429 https://chan.sankakucomplex.com/post/show/85992  
  25. [L38] 429 https://chan.sankakucomplex.com/post/show/9482  
  26. [L39] 429 https://chan.sankakucomplex.com/post/show/9472  
  27. [L40] 429 https://chan.sankakucomplex.com/post/show/949242  
  28. [L41] 429 https://chan.sankakucomplex.com/post/show/96562  
  29. [L42] 429 https://chan.sankakucomplex.com/post/show/992333  
  30. [L43] 429 https://chan.sankakucomplex.com/post/show/9053  
  31. [L45] 429 https://chan.sankakucomplex.com/post/show/82138  
  32. [L46] 429 https://chan.sankakucomplex.com/post/show/821541  
  33. [L47] 429 https://chan.sankakucomplex.com/post/show/841569  
  34. [L48] 429 https://chan.sankakucomplex.com/post/show/851296  
  35. [L49] 429 https://chan.sankakucomplex.com/post/show/941883  
  36. [L50] 429 https://chan.sankakucomplex.com/post/show/941875  
  37. [L51] 429 https://chan.sankakucomplex.com/post/show/941241  
  38. [L52] 429 https://chan.sankakucomplex.com/post/show/961668  
  39. [L53] 429 https://chan.sankakucomplex.com/post/show/991332  
  40. [L54] 429 https://chan.sankakucomplex.com/post/show/991051  
> Dupes 
  None ✓
```

With fix, run time 0m8.782s

```
./awesome_bot max-req 
> Checking links in max-req
Links to check: 50
  01. https://chan.sankakucomplex.com/post/show/82338 
  02. https://chan.sankakucomplex.com/post/show/828541 
  03. https://chan.sankakucomplex.com/post/show/847569 
  04. https://chan.sankakucomplex.com/post/show/859296 
  05. https://chan.sankakucomplex.com/post/show/945883 
  06. https://chan.sankakucomplex.com/post/show/947875 
  07. https://chan.sankakucomplex.com/post/show/949241 
  08. https://chan.sankakucomplex.com/post/show/965668 
  09. https://chan.sankakucomplex.com/post/show/992332 
  10. https://chan.sankakucomplex.com/post/show/998051 
  11. https://chan.sankakucomplex.com/post/show/82323 
  12. https://chan.sankakucomplex.com/post/show/82841 
  13. https://chan.sankakucomplex.com/post/show/84769 
  14. https://chan.sankakucomplex.com/post/show/829296 
  15. https://chan.sankakucomplex.com/post/show/925883 
  16. https://chan.sankakucomplex.com/post/show/927875 
  17. https://chan.sankakucomplex.com/post/show/929241 
  18. https://chan.sankakucomplex.com/post/show/925668 
  19. https://chan.sankakucomplex.com/post/show/922332 
  20. https://chan.sankakucomplex.com/post/show/928051 
  21. https://chan.sankakucomplex.com/post/show/8233 
  22. https://chan.sankakucomplex.com/post/show/8285 
  23. https://chan.sankakucomplex.com/post/show/8475 
  24. https://chan.sankakucomplex.com/post/show/8592 
  25. https://chan.sankakucomplex.com/post/show/9458 
  26. https://chan.sankakucomplex.com/post/show/9478 
  27. https://chan.sankakucomplex.com/post/show/9492 
  28. https://chan.sankakucomplex.com/post/show/9656 
  29. https://chan.sankakucomplex.com/post/show/9923 
  30. https://chan.sankakucomplex.com/post/show/9980 
  31. https://chan.sankakucomplex.com/post/show/8281 
  32. https://chan.sankakucomplex.com/post/show/8542 
  33. https://chan.sankakucomplex.com/post/show/847562 
  34. https://chan.sankakucomplex.com/post/show/85992 
  35. https://chan.sankakucomplex.com/post/show/9482 
  36. https://chan.sankakucomplex.com/post/show/9472 
  37. https://chan.sankakucomplex.com/post/show/949242 
  38. https://chan.sankakucomplex.com/post/show/96562 
  39. https://chan.sankakucomplex.com/post/show/992333 
  40. https://chan.sankakucomplex.com/post/show/9053 
  41. https://chan.sankakucomplex.com/post/show/82138 
  42. https://chan.sankakucomplex.com/post/show/821541 
  43. https://chan.sankakucomplex.com/post/show/841569 
  44. https://chan.sankakucomplex.com/post/show/851296 
  45. https://chan.sankakucomplex.com/post/show/941883 
  46. https://chan.sankakucomplex.com/post/show/941875 
  47. https://chan.sankakucomplex.com/post/show/941241 
  48. https://chan.sankakucomplex.com/post/show/961668 
  49. https://chan.sankakucomplex.com/post/show/991332 
  50. https://chan.sankakucomplex.com/post/show/991051 
Checking URLs: ✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓
No issues :-)
```

**Comment by dkhamsing** [comment]
will have to work on this more later
keep u posted

**Comment by dkhamsing** [comment]
@soredake the fix is to add a request delay option
should work

what do you think

**Comment by dkhamsing** [comment]
New feature to add request delay, published in `1.14.0` today

**Comment by soredake** [comment]
Looks good, I'll test it soon.
